agent/src/AgentTextEditor.ts

```diff
@@ -4,23 +4,35 @@ import type { AgentTextDocument } from './AgentTextDocument'
 import type { EditFunction } from './AgentWorkspaceDocuments'
 
 export class AgentTextEditor implements vscode.TextEditor {
+    private _selection: vscode.Selection
     constructor(
         private readonly agentDocument: AgentTextDocument,
         private readonly params?: { edit?: EditFunction }
-    ) {}
+    ) {
+        this._selection = new vscode.Selection(new vscode.Position(0, 0), new vscode.Position(0, 0))
+        const protocolSelection = this.agentDocument.protocolDocument.selection
+        if (protocolSelection) {
+            this._selection = new vscode.Selection(
+                new vscode.Position(protocolSelection.start.line, protocolSelection.start.character),
+                new vscode.Position(protocolSelection.end.line, protocolSelection.end.character)
+            )
+        }
+    }
     get document(): AgentTextDocument {
         return this.agentDocument
     }
     get selection(): vscode.Selection {
         const protocolSelection = this.agentDocument.protocolDocument.selection
-        const selection: vscode.Selection = protocolSelection
-            ? new vscode.Selection(
-                  new vscode.Position(protocolSelection.start.line, protocolSelection.start.character),
-                  new vscode.Position(protocolSelection.end.line, protocolSelection.end.character)
-              )
-            : // Default to putting the cursor at the start of the file.
-              new vscode.Selection(new vscode.Position(0, 0), new vscode.Position(0, 0))
-        return selection
+        if (protocolSelection) {
+            this._selection = new vscode.Selection(
+                new vscode.Position(protocolSelection.start.line, protocolSelection.start.character),
+                new vscode.Position(protocolSelection.end.line, protocolSelection.end.character)
+            )
+        }
+        return this._selection
+    }
+    set selection(newSelection: vscode.Selection) {
+        this._selection = newSelection
```

Comment:
```suggestion
        // Do nothing as selection is set by agentDocument
```

I think we didn't implement the set selection() because it's not needed, so we can just add a no-op method to bypass the error `Error: Cannot set property selection of #<AgentTextEditor> which has only a getter` as we don't need to set the selection twice in TextEditor and AgentDocument

Reply:
The test is still passing after making all the suggested changes: 
![image](https://github.com/user-attachments/assets/ad432ace-40f9-4a69-aebf-0d9223683349)